core/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! # Subxt-core
+//!
+//! `#[no_std]` compatible core crate for subxt.
+
```

Comment:
we probably want some more lints here such as 

```
#![cfg_attr(docsrs, feature(doc_cfg))]
```

to work with the macros you added in core/src/macros.rs

Reply:
IMO we should have `#![deny(missing_docs)]` everywhere :)